### PR TITLE
Change avahi test to pytest.

### PR DESCRIPTION
**Pull request by petedyerarm**
Changed the avahi test to run as a pytest.
Added timeout parameter to the execute_command method. The default is
set to no timeout.
This allows for recovery of a hanging avahi-browse command which was
previously performed in the shell script.

### CODE REVIEW

**Comment by petedyerarm** [comment]
synchronize with https://github.com/ARMmbed/mbl-tools/pull/352

File: ci/lava/tests/test-avahi-discovery.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+"""Script to run avahi discovery test."""
+
+from datetime import datetime
+import time
+
+
+class TestRunner:
+    """Class to encapsulate the avahi discover of a DUT."""
+
+    def _test_avahi_browse(self, execute_helper):
```
Comment (diegorusso) [inline ci/lava/tests/test-avahi-discovery.py]: Can we have this method at the end of the class?

Comment (diegorusso) [reply ci/lava/tests/test-avahi-discovery.py]: Can we also rename it to something else? (_run_avahi_browse, _discover_ssh_service ?)

File: ci/lava/tests/test-avahi-discovery.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        returnValue = False
+
+        test_command = ["avahi-browse", "-tr", "_ssh._tcp"]
+        err, device_list, error = execute_helper.execute_command(
+            test_command, 60
+        )
+
+        if err == 0:
+            lines = device_list.split("\n")
```
Comment (diegorusso) [inline ci/lava/tests/test-avahi-discovery.py]: How does a regex look like for this?

File: ci/lava/tests/test-avahi-discovery.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        return returnValue
+
+    def test_avahi_browse(self, execute_helper):
+        """Perform the avahi browse test.
+
+        Repeatedly perform avahi-browse calls, one per second, until either a
+        device is found or 5 minutes have elapsed.
+        """
+        returnValue = False
+        starttime = datetime.now()
```
Comment (diegorusso) [inline ci/lava/tests/test-avahi-discovery.py]:
```suggestion
        start_time = datetime.now()
```

File: ci/lava/tests/test-avahi-discovery.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        while (
+            not returnValue and (datetime.now() - starttime).seconds < 5 * 60
```
Comment (diegorusso) [inline ci/lava/tests/test-avahi-discovery.py]:
```suggestion
            not returnValue and (datetime.now() - start_time).seconds < 5 * 60
```

Comment (diegorusso) [reply ci/lava/tests/test-avahi-discovery.py]: This form is a bit cryptic and hard to read. Can we do better?

File: ci/lava/tests/test-avahi-discovery.py (hunk 1 above)
Comment (petedyerarm) [reply ci/lava/tests/test-avahi-discovery.py]: renamed to _run_avahi_browse and moved to the end of the class.

**Pull request merged**